### ISSUE TITLE
"param" shortcode is not working in some cases

### ISSUE THREAD

**Issue by jihoon-seo** [kubernetes/website#35686]
**This is a Bug Report**

**Problem:**
"param" shortcode is not working in some cases.

https://github.com/kubernetes/website/blob/main/content/en/docs/tasks/administer-cluster/kubeadm/upgrading-windows-nodes.md?plain=1

https://github.com/kubernetes/website/blob/7ae76503960e5dd735ce02caa4fd16fc502c2464/content/en/docs/tasks/administer-cluster/kubeadm/upgrading-windows-nodes.md?plain=1#L35-L38

![image](https://user-images.githubusercontent.com/46767780/182741842-0a8f872b-721c-40e1-93d7-0c4c9198efdf.png)

https://kubernetes.io/docs/tasks/administer-cluster/kubeadm/upgrading-windows-nodes/

![image](https://user-images.githubusercontent.com/46767780/182742022-61b072b7-630f-4199-a6fc-889a040380b9.png)

**Proposed Solution:**

**Page to Update:**
- https://kubernetes.io/docs/tasks/administer-cluster/kubeadm/upgrading-windows-nodes/#:~:text=https%3A//dl.k8s.io//bin/windows/amd64/kubeadm.exe
- ...

**Comment by Shubham82** [comment]
Hi @jihoon-seo 
I think it is due to the comment mentioned above the URL i.e. `# replace v1.24.0 with your desired version`, so if we remove that then it will show the version in the URL.

> curl.exe -Lo <path-to-kubeadm.exe>  https://dl.k8s.io/{{< param "fullversion" >}}/bin/windows/amd64/kubeadm.exe

`curl.exe -Lo <path-to-kubeadm.exe>  https://dl.k8s.io/v1.24.0/bin/windows/amd64/kubeadm.exe
`

or another solution is to enclose the URL in inverted commas (" "), it will show the full version in the URL.  I tried and following below is the output:
> curl.exe -Lo <path-to-kubeadm.exe>  "https://dl.k8s.io/{{< param "fullversion" >}}/bin/windows/amd64/kubeadm.exe"

```
# replace v1.24.0 with your desired version
curl.exe -Lo <path-to-kubeadm.exe>  "https://dl.k8s.io/v1.24.0/bin/windows/amd64/kubeadm.exe"
```

and also adding  one more # to the comment also show the version:

   ```
## replace {{< param "fullversion" >}} with your desired version
    curl.exe -Lo <path-to-kubeadm.exe>  https://dl.k8s.io/{{< param "fullversion" >}}/bin/windows/amd64/kubeadm.exe

```
```
## replace v1.24.0 with your desired version
curl.exe -Lo <path-to-kubeadm.exe>  https://dl.k8s.io/v1.24.0/bin/windows/amd64/kubeadm.exe

```

**Comment by Shubham82** [comment]
What do you think is the best solution for it?
IMO it would be  `enclose the URL in inverted commas (" "), it will show the full version in the URL`

**Comment by jihoon-seo** [comment]
@Shubham82 thank you for the research!
It seems somehow weird for me that the line `# replace {{< param "fullversion" >}} with your desired version` with single `#` causes this problem, and that removing that line or adding one more `#` resolves the problem.

Among the options you suggested, I also prefer the option you chosen: `enclose the URL in inverted commas (" "), it will show the full version in the URL.`

**Comment by Shubham82** [comment]
Hi @jihoon-seo 
When I tried to build it locally without any changes, it will show the version in the URL and also in [Upgrade kubelet and kube-proxy](https://kubernetes.io/docs/tasks/administer-cluster/kubeadm/upgrading-windows-nodes/#upgrade-kubelet-and-kube-proxy) the steps mentioned is not correct as per the code.

mentioned in [upgrading-windows-nodes.md](https://github.com/kubernetes/website/blob/main/content/en/docs/tasks/administer-cluster/kubeadm/upgrading-windows-nodes.md) 
```
1.  From the Windows node, upgrade and restart the kubelet:

    ```powershell
    stop-service kubelet
    curl.exe -Lo <path-to-kubelet.exe> https://dl.k8s.io/{{< param "fullversion" >}}/bin/windows/amd64/kubelet.exe
    restart-service kubelet
    ```

2. From the Windows node, upgrade and restart the kube-proxy.

    ```powershell
    stop-service kube-proxy
    curl.exe -Lo <path-to-kube-proxy.exe> https://dl.k8s.io/{{< param "fullversion" >}}/bin/windows/amd64/kube-proxy.exe
    restart-service kube-proxy
    ```
```
output: restart-service kubelet and  restart-service kube-proxy append to the URL. also Param shortcode is also not working here.

```
stop-service kubelet
curl.exe -Lo <path-to-kubelet.exe> https://dl.k8s.io//bin/windows/amd64/kubelet.exerestart-service kubelet
```

 ```
 stop-service kube-proxy
curl.exe -Lo <path-to-kube-proxy.exe> https://dl.k8s.io//bin/windows/amd64/kube-proxy.exerestart-service kube-proxy
```

But when you build it locally using  [Running the website locally using Hugo](https://github.com/kubernetes/website#running-the-website-locally-using-hugo) This page works fine for me. No above error is there. it seems weird to me. could you please try to build locally without any changes?

**Comment by jihoon-seo** [comment]
@Shubham82 

[Prerequisites' versions in my dev env]
```
↳ node --version
v16.15.1

↳ npm --version 
8.11.0

↳ go version
go version go1.18.3 linux/amd64

↳ hugo version
hugo v0.101.0-466fa43c16709b4483689930a4f9ac8add5c9f66+extended linux/amd64 BuildDate=2022-06-16T07:09:16Z

↳ git submodule update --init --recursive --depth 1
# succeeded with no output
```

[Running the website locally using Hugo]
```
↳ npm ci
npm WARN old lockfile 
npm WARN old lockfile The package-lock.json file was created with an old version of npm,
npm WARN old lockfile so supplemental metadata must be fetched from the registry.
npm WARN old lockfile 
npm WARN old lockfile This is a one-time fix-up, please be patient...
npm WARN old lockfile 
npm WARN deprecated core-js-pure@3.6.5: core-js-pure@<3.23.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Some versions have web compatibility issues. Please, upgrade your dependencies to the actual version of core-js-pure.

added 119 packages, and audited 120 packages in 2s

9 packages are looking for funding
  run `npm fund` for details

6 vulnerabilities (2 moderate, 4 high)

To address all issues, run:
  npm audit fix

Run `npm audit` for details.
```

```
↳ make serve   
hugo server --buildFuture --environment development
Start building sites … 
hugo v0.101.0-466fa43c16709b4483689930a4f9ac8add5c9f66+extended linux/amd64 BuildDate=2022-06-16T07:09:16Z
WARNING: calling IsSet with unsupported type "invalid" (<nil>) will always return false.
Makefile:54: recipe for target 'serve' failed
```

I cannot find why this fails.

---

[Running the website using a container]
```
↳ make container-serve
"docker" run --rm --interactive --tty --volume "/home/jhseo/k8s-website:/src" --cap-drop=ALL --cap-add=AUDIT_WRITE --read-only --mount type=tmpfs,destination=/tmp,tmpfs-mode=01777 -p 1313:1313 gcr.io/k8s-staging-sig-docs/k8s-website-hugo:v0.101.0-215aaa9df947 hugo server --buildFuture --environment development --bind 0.0.0.0 --destination /tmp/hugo --cleanDestinationDir --noBuildLock
Start building sites … 
hugo v0.101.0+extended linux/amd64 BuildDate=unknown
WARNING: calling IsSet with unsupported type "invalid" (<nil>) will always return false.

                   |  EN  | ZH-CN | KO  | JA  | FR  | IT  | DE  | ES  | PT-BR | ID  | VI  | RU  | PL  | UK   
-------------------+------+-------+-----+-----+-----+-----+-----+-----+-------+-----+-----+-----+-----+------
  Pages            | 1463 |  1104 | 543 | 456 | 338 |  78 | 164 | 294 |   217 | 335 |  77 | 155 |  69 |  92  
  Paginator pages  |   44 |    10 |   0 |   0 |   0 |   0 |   0 |   0 |     0 |   0 |   0 |   0 |   0 |   0  
  Non-page files   |  513 |   399 | 205 | 265 |  73 |  20 |  17 |  39 |    30 | 105 |   8 |  24 |   6 |  20  
  Static files     |  837 |   837 | 837 | 837 | 837 | 837 | 837 | 837 |   837 | 837 | 837 | 837 | 837 | 837  
  Processed images |    1 |     1 |   0 |   0 |   0 |   0 |   0 |   0 |     0 |   0 |   0 |   0 |   0 |   0  
  Aliases          |    8 |     6 |   3 |   1 |   0 |   1 |   0 |   0 |     1 |   1 |   0 |   1 |   0 |   0  
  Sitemaps         |    2 |     1 |   1 |   1 |   1 |   1 |   1 |   1 |     1 |   1 |   1 |   1 |   1 |   1  
  Cleaned          |    0 |     0 |   0 |   0 |   0 |   0 |   0 |   0 |     0 |   0 |   0 |   0 |   0 |   0  

Built in 11025 ms
Watching for changes in /src/{archetypes,assets,content,data,i18n,layouts,package.json,postcss.config.js,static,themes}
Watching for config changes in /src/config.toml, /src/themes/docsy/config.toml, /src/go.mod
Environment: "development"
Serving pages from /tmp/hugo
Running in Fast Render Mode. For full rebuilds on change: hugo server --disableFastRender
Web Server is available at http://localhost:1313/ (bind address 0.0.0.0)
Press Ctrl+C to stop
```

![image](https://user-images.githubusercontent.com/46767780/182809592-3f3e96c9-fc2d-46b8-b2a7-1839ca987b30.png)

The problem persists.

**Comment by Shubham82** [comment]
Hi, @jihoon-seo I tried both methods to build the website locally, and both give different outputs, with [Running the website using a container](https://github.com/kubernetes/website#running-the-website-using-a-container) the problem still persists. below are the following screenshots of that:

![Hugo-Screenshot3](https://user-images.githubusercontent.com/50987400/182836466-a867b968-e7ab-4757-ade4-c65b3897899b.png)

![Hugo-Screenshot4](https://user-images.githubusercontent.com/50987400/182836458-e6f664b6-39af-4ab1-9ace-cc1b72b94a98.png)

but with the [Running the website locally using Hugo](https://github.com/kubernetes/website#running-the-website-locally-using-hugo) it works fine for me, please find screenshots of it below:


![Hugo-Screenshot1](https://user-images.githubusercontent.com/50987400/182836808-67516596-4fa0-4464-86d3-7d19eb8b056e.png)

![Hugo-Screenshot2](https://user-images.githubusercontent.com/50987400/182836801-29abcaff-7da5-40cf-9dcd-f1ba1d0d2ed6.png)

Don't know why it is happening.

**Comment by jihoon-seo** [comment]
@Shubham82 Hmm, I can see that the Hugo version you are using (v0.76.5) is somewhat old, compared to what Netlify is using to build `k8s.io`.

https://github.com/kubernetes/website/blob/main/netlify.toml
https://github.com/kubernetes/website/blob/9ee4245ccf3b7401f20471ab0c46696ded055c64/netlify.toml#L10-L12

Could you try upgrading your Hugo, and then check whether the error persists or not? 😊

**Comment by Shubham82** [comment]
Sure I update Hugo and try it again!

**Comment by sftim** [comment]
/remove-sig windows
/remove-sig cluster-lifecycle

Nothing for folks from those SIGs to fix.

**Comment by Shubham82** [comment]
Hi @jihoon-seo 
I confirmed that on updating Hugo to `v0.101.0` the error is also reproducible with the [Running the website locally using Hugo](https://github.com/kubernetes/website#running-the-website-locally-using-hugo).

Thanks!

**Comment by Shubham82** [comment]
Another solution for it is if we write the text under the `shell` block instead of  `powershell`  will resolve this. please see this:

```
```shell
    # replace {{< param "fullversion" >}} with your desired version
    curl.exe -Lo <path-to-kubeadm.exe>  https://dl.k8s.io/{{< param "fullversion" >}}/bin/windows/amd64/kubeadm.exe
    ```
```
![Screenshot1](https://user-images.githubusercontent.com/50987400/183019081-291f4f46-9ce5-4dcc-a839-d70f93aae05f.png)


```
```shell
    stop-service kubelet
    curl.exe -Lo <path-to-kubelet.exe> https://dl.k8s.io/{{< param "fullversion" >}}/bin/windows/amd64/kubelet.exe
    restart-service kubelet
    ```

```shell
    stop-service kube-proxy
    curl.exe -Lo <path-to-kube-proxy.exe> https://dl.k8s.io/{{< param "fullversion" >}}/bin/windows/amd64/kube-proxy.exe
    restart-service kube-proxy
    ```
```
![Screenshot2](https://user-images.githubusercontent.com/50987400/183019101-c2265076-4198-4319-bb40-d91929c89d36.png)

**Comment by Shubham82** [comment]
if it seems good then I will open a PR for it!

**Comment by jihoon-seo** [comment]
@Shubham82 it lgtm, please do so!
/assign @Shubham82

**Comment by Shubham82** [comment]
Fixed it here: #35725
